src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1692 @@
+      N_Vector* ptrtempv1 = &(ark_mem->tempv1);
+      N_Vector* ptrtempv2 = &(ark_mem->tempv2);
+
+      N_Vector temp = *ptrtempv2;
+      *ptrtempv2    = *ptrtempv1;
+      *ptrtempv1    = temp;
+
+      N_VScale(ONE, ark_mem->ycur, ark_mem->tempv2);
+
+      cjm1 = cj;
+      bjm2 = bjm1;
+      bjm1 = bj;
+    }
+  }
+
+  /* Compute yerr (if step adaptivity enabled) */
+  if (!ark_mem->fixedstep)
+  {
+    retval = step_mem->fe(ark_mem->tcur + ark_mem->h, ark_mem->ycur,
+                          ark_mem->tempv2, ark_mem->user_data);
+    step_mem->nfe++;
+    if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+
+    /* Estimate the local error and compute its weighted RMS norm */
+    cvals[0] = p8;
+    Xvecs[0] = ark_mem->yn;
+    cvals[1] = -p8;
+    Xvecs[1] = ark_mem->ycur;
+    cvals[2] = p4 * ark_mem->h;
+    Xvecs[2] = ark_mem->fn;
+    cvals[3] = p4 * ark_mem->h;
+    Xvecs[3] = ark_mem->tempv2;
+
+    retval = N_VLinearCombination(4, cvals, Xvecs, ark_mem->tempv1);
+    if (retval != 0) { return (ARK_VECTOROP_ERR); }
+
+    *dsmPtr = N_VWrmsNorm(ark_mem->tempv1, ark_mem->ewt);
+    lsrkStep_DomEigUpdateLogic(ark_mem, step_mem, *dsmPtr);
+  }
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_TakeStepSSPs2:
+
+  This routine serves the primary purpose of the LSRKStepSSPs2 module:
+  it performs a single SSPs2 step (with embedding).
+
+  The output variable dsmPtr should contain estimate of the
+  weighted local error if an embedding is present; otherwise it
+  should be 0.
+
+  The input/output variable nflagPtr is used to gauge convergence
+  of any algebraic solvers within the step.  As this routine
+  involves no algebraic solve, it is set to 0 (success).
+
+  The return value from this routine is:
+            0 => step completed successfully
+           >0 => step encountered recoverable failure;
+                 reduce step and retry (if possible)
+           <0 => step encountered unrecoverable failure
+  ---------------------------------------------------------------*/
+int lsrkStep_TakeStepSSPs2(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
+{
+  int retval;
+
+  ARKodeLSRKStepMem step_mem;
+
+  /* initialize algebraic solver convergence flag to success,
+     temporal error estimate to zero */
+  *nflagPtr = ARK_SUCCESS;
+  *dsmPtr   = ZERO;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  sunrealtype* cvals;
+  N_Vector* Xvecs;
+
+  cvals = step_mem->cvals;
+  Xvecs = step_mem->Xvecs;
+
+  sunrealtype rs     = (sunrealtype)step_mem->req_stages;
+  sunrealtype sm1inv = ONE / (rs - ONE);
+  sunrealtype bt1, bt2, bt3;
+
+  /* Embedding coefficients differ when req_stages == 2 */
+  if (step_mem->req_stages == 2)
+  {
+    bt1 = SUN_RCONST(0.694021459207626);
```

Comment:
If I understand correctly, this embedding is order 1 so the only order condition here is the b's sum up to 1. This seems to be the case even when using extended precision. Adding more digits, while nice if available, will have negligible impact. I essentially had the same situation with the [Shu-Osher method](https://sundials.readthedocs.io/en/latest/arkode/Butcher_link.html#shu-osher-3-2-3), although that SSP method has a second order embedding.

It would be good to comment that this "magic constant" comes from https://doi.org/10.1016/j.cam.2022.114325 pg 5